File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:

        i = 0
        j = len(height)-1
        area = 0

        while (i<j):
            area = max(area, min(height[i], height[j])*(j-i))
            if(height[i]<height[j]):
                i+=1
            else:
                j-=1
        
        return area


        

File: 11-container-with-most-water/container-with-most-water.java

class Solution {
    public int maxArea(int[] height) {
        int area = 0;
        
        int l = 0;
        int r = height.length - 1;

        while (l < r) {
            int width = r - l;
            
            int currentArea = Math.min(height[l], height[r]) * width;
            area = Math.max(area, currentArea);
            
            if (height[l] <= height[r]) {
                l++;
            } else {
                r--;
            }
        }
        
        return area;
    }
}

